Software/src/UpdatesProcessor.cpp

```diff
@@ -60,38 +61,39 @@ void UpdatesProcessor::requestUpdates()
 		_reply = NULL;
 	}
 
-	QNetworkRequest request(QUrl(UPDATE_CHECK_URL));
+	QNetworkRequest request(QUrl(QStringLiteral(UPDATE_CHECK_URL)));
+	request.setHeader(QNetworkRequest::UserAgentHeader, QVariant("psieg/Lightpack"));
 	request.setSslConfiguration(QSslConfiguration::defaultConfiguration());
 	_reply = _networkMan.get(request);
 	connect(_reply, SIGNAL(finished()), this, SIGNAL(readyRead()));
 	connect(_reply, SIGNAL(error(QNetworkReply::NetworkError)), this, SLOT(error(QNetworkReply::NetworkError)));
 }
 
-QList<UpdateInfo> UpdatesProcessor::readUpdates()
+UpdateInfo UpdatesProcessor::readUpdates()
 {
-	QList<UpdateInfo> updates;
-
-	QXmlStreamReader xmlReader(_reply->readAll());
-
-	if (xmlReader.readNextStartElement()) {
-		if (xmlReader.name() == "updates") {
-			readUpdates(&updates, &xmlReader);
-			QList<UpdateInfo>::iterator it = updates.begin();
-			while (it != updates.end()) {
-				UpdateInfo updateInfo = *it;
-				if (!updateInfo.softwareVersion.isEmpty() && !isVersionMatches(updateInfo.softwareVersion, kCurVersion)) {
-					updates.removeAll(updateInfo);
-				} else {
-					it++;
-				}
-			}
-		}
+	const QJsonObject& releaseObject = QJsonDocument::fromJson(_reply->readAll()).object();
+
+	UpdateInfo update;
+
+	update.id = releaseObject.value(QStringLiteral("id")).toInt();
+	update.url = releaseObject.value(QStringLiteral("html_url")).toString();
+	update.text = releaseObject.value(QStringLiteral("body")).toString();
+	update.title = releaseObject.value(QStringLiteral("name")).toString();
+	update.softwareVersion = QVersionNumber::fromString(releaseObject.value(QStringLiteral("tag_name")).toString());
```

Comment:
What if this fails, e.g. "FW_version_6+1"?

Reply:
`softwareVersion` will be `isNull()` internally and `compare()` will treat it as v`0`, so no update